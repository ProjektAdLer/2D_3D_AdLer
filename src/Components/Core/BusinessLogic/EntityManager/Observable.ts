export default abstract class Observable<T> {
  protected value: T;
  protected subscribers: ((value: T) => void)[] = [];

  public subscribe(fn: (value: T) => void): void {
    this.subscribers.push(fn);
  }

  public unsubscribe(fn: (value: T) => void): void {
    this.subscribers = this.subscribers.filter((subscriber) => {
      return subscriber !== fn;
    });
  }

  protected notify(data: T): void {
    this.subscribers.forEach((fn) => {
      fn(data);
    });
  }

  get Value(): T {
    return this.value;
  }
}
