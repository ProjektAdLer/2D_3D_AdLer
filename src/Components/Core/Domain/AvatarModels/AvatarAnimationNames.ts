const AvatarAnimationNames = {
  // animations for avatar
  idle: "ac_anim_idle2_avatarSkeleton",
  walk: "ac_anim_walkcycle_avatarSkeleton",
  interact: "ac_anim_interact_avatarSkeleton",
  // animations for npcs
  npc_idle: "ac_anim_idle2",
  npc_walk: "ac_anim_walkcycle",
} as const;

export default AvatarAnimationNames;
