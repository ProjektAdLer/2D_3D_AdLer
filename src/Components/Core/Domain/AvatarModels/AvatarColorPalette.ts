/**
 * Represents a color in the avatar color palette.
 */
export interface AvatarColor {
  /**
   * The id of the color on the color palette.
   */
  id: number;

  /**
   * The key for the name of the color, used for localization.
   */
  nameKey: string;

  /**
   * The hexadecimal representation of the color.
   */
  hexColor: string;
}

const AvatarColorPalette: AvatarColor[] = [
  // Black
  { id: 0, nameKey: "Black 1", hexColor: "#000000" },
  { id: 1, nameKey: "Black 2", hexColor: "#1b1b1b" },
  { id: 2, nameKey: "Black 3", hexColor: "#313131" },
  { id: 3, nameKey: "Black 4", hexColor: "#555555" },
  { id: 4, nameKey: "Black 5", hexColor: "#767676" },
  { id: 5, nameKey: "Black 6", hexColor: "#949494" },
  { id: 6, nameKey: "Black 7", hexColor: "#b3b3b3" },
  { id: 7, nameKey: "Black 8", hexColor: "#cecece" },
  // Blue
  { id: 8, nameKey: "Blue 1", hexColor: "#043675" },
  { id: 9, nameKey: "Blue 2", hexColor: "#043675" }, // duplicate
  { id: 10, nameKey: "Blue 3", hexColor: "#2f91cd" },
  { id: 11, nameKey: "Blue 4", hexColor: "#83aece" },
  { id: 12, nameKey: "Blue 5", hexColor: "#17607f" },
  { id: 13, nameKey: "Blue 6", hexColor: "#5398b0" },
  { id: 14, nameKey: "Blue 7", hexColor: "#94b7c3" },
  { id: 15, nameKey: "Blue 8", hexColor: "#b4c6ce" },
  // Red
  { id: 16, nameKey: "Red 1", hexColor: "#8b1518" },
  { id: 17, nameKey: "Red 2", hexColor: "#a41e22" },
  { id: 18, nameKey: "Red 3", hexColor: "#bd2c30" },
  { id: 19, nameKey: "Red 4", hexColor: "#ce4145" },
  { id: 20, nameKey: "Red 5", hexColor: "#871a0b" },
  { id: 21, nameKey: "Red 6", hexColor: "#a8411e" },
  { id: 22, nameKey: "Red 7", hexColor: "#c1643f" },
  { id: 23, nameKey: "Red 8", hexColor: "#cdada2" },
  // Orange
  { id: 24, nameKey: "Orange 1", hexColor: "#b76000" },
  { id: 25, nameKey: "Orange 2", hexColor: "#c98300" },
  { id: 26, nameKey: "Orange 3", hexColor: "#cea100" },
  { id: 27, nameKey: "Orange 4", hexColor: "#ceb753" },
  { id: 28, nameKey: "Orange 5", hexColor: "#aa9501" },
  { id: 29, nameKey: "Orange 6", hexColor: "#c2af09" },
  { id: 30, nameKey: "Orange 7", hexColor: "#ccc428" },
  { id: 31, nameKey: "Orange 8", hexColor: "#cccaa5" },
  // Brown
  { id: 32, nameKey: "Brown 1", hexColor: "#25160c" },
  { id: 33, nameKey: "Brown 2", hexColor: "#4b2a1a" },
  { id: 34, nameKey: "Brown 3", hexColor: "#633d29" },
  { id: 35, nameKey: "Brown 4", hexColor: "#835b32" },
  { id: 36, nameKey: "Brown 5", hexColor: "#bb9871" },
  { id: 37, nameKey: "Brown 6", hexColor: "#a6947d" },
  { id: 38, nameKey: "Brown 7", hexColor: "#b1a28b" },
  { id: 39, nameKey: "Brown 8", hexColor: "#bba989" },
  // more Brown
  { id: 40, nameKey: "Brown 9", hexColor: "#7d4337" },
  { id: 41, nameKey: "Brown 10", hexColor: "#b97947" },
  { id: 42, nameKey: "Brown 11", hexColor: "#ac7d59" },
  { id: 43, nameKey: "Brown 12", hexColor: "#677f88" },
  { id: 44, nameKey: "Brown 13", hexColor: "#b29676" },
  { id: 45, nameKey: "Brown 14", hexColor: "#bfa183" },
  { id: 46, nameKey: "Brown 15", hexColor: "#c9b2a6" },
  { id: 47, nameKey: "Brown 16", hexColor: "#ccbeb8" },
  // violet
  { id: 48, nameKey: "Violet 1", hexColor: "#a909b3" },
  { id: 49, nameKey: "Violet 2", hexColor: "#c55eca" },
  { id: 50, nameKey: "Violet 3", hexColor: "#c8a6cb" },
  { id: 51, nameKey: "Violet 4", hexColor: "#cbc0cc" },
  { id: 52, nameKey: "Violet 5", hexColor: "#500c8c" },
  { id: 53, nameKey: "Violet 6", hexColor: "#863fb3" },
  { id: 54, nameKey: "Violet 7", hexColor: "#a370c3" },
  { id: 55, nameKey: "Violet 8", hexColor: "#bdaacb" },
  // green
  { id: 56, nameKey: "Green 1", hexColor: "#0d5502" },
  { id: 57, nameKey: "Green 2", hexColor: "#199a09" },
  { id: 58, nameKey: "Green 3", hexColor: "#3ac61e" },
  { id: 59, nameKey: "Green 4", hexColor: "#82ce84" },
  { id: 60, nameKey: "Green 5", hexColor: "#3b6613" },
  { id: 61, nameKey: "Green 6", hexColor: "#709636" },
  { id: 62, nameKey: "Green 7", hexColor: "#9db16b" },
  { id: 63, nameKey: "Green 8", hexColor: "#bdc6a5" },
];

export default AvatarColorPalette;
