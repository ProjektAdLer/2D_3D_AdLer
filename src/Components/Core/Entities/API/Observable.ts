export default class Observable<T> {
  constructor(data?: T) {
    if (data) {
      this.data = data;
    }
  }
  private data: T;
  private subscribers: Array<(data: T) => void> = [];

  public subscribe(fn: (data: T) => void): void {
    this.subscribers.push(fn);
  }

  public unsubscribe(fn: (data: T) => void): void {
    this.subscribers = this.subscribers.filter((subscriber) => {
      return subscriber !== fn;
    });
  }

  public notify(data: T): void {
    this.subscribers.forEach((fn) => {
      fn(data);
    });
  }

  public getData(): T {
    return this.data;
  }

  public setData(data: T): void {
    this.data = data;
    this.notify(data);
  }
}
