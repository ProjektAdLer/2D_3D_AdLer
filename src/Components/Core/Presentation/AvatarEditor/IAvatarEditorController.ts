import AvatarConfigTO from "../../Application/DataTransferObjects/AvatarConfigTO";

export default interface IAvatarEditorController {
  onAvatarConfigChanged(changes: Partial<AvatarConfigTO>): void;
  saveAvatarConfig(): void;
  resetAvatarConfig(): void;
}
