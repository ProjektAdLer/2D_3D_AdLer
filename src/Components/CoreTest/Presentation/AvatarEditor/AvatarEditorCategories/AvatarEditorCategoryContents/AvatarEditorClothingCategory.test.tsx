import { mock } from "jest-mock-extended";
import AvatarEditorViewModel, {
  AvatarEditorUI,
} from "../../../../../Core/Presentation/AvatarEditor/AvatarEditorViewModel";
import IAvatarEditorController from "../../../../../Core/Presentation/AvatarEditor/IAvatarEditorController";
import Observable from "../../../../../../Lib/Observable";
import {
  AvatarPantsModels,
  AvatarShirtModels,
  AvatarShoesModels,
} from "../../../../../Core/Domain/AvatarModels/AvatarModelTypes";
import { AvatarColor } from "../../../../../Core/Domain/AvatarModels/AvatarColorPalette";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import AvatarEditorClothingCategory from "../../../../../Core/Presentation/AvatarEditor/AvatarEditorCategories/AvatarEditorCategoryContents/AvatarEditorClothingCategory";

const avatarEditorMock = mock<AvatarEditorViewModel>();
const avatarEditorControllerMock = mock<IAvatarEditorController>();

describe("AvatarEditorClothingCategory", () => {
  // ANF-ID: [EZZ0042, EZZ0043, EZZ0044, EZZ0045, EZZ0046, EZZ0047]
  test("should render", () => {
    avatarEditorMock.uiVisiblity = {
      clothingMenu: {
        shirts: new Observable<boolean>(true),
        pants: new Observable<boolean>(true),
        shoes: new Observable<boolean>(true),
      },
    } as AvatarEditorUI;

    avatarEditorMock.shirt = new Observable<AvatarShirtModels>("none" as any);
    avatarEditorMock.pants = new Observable<AvatarPantsModels>("none" as any);
    avatarEditorMock.shoes = new Observable<AvatarShoesModels>("none" as any);
    avatarEditorMock.shirtColor = new Observable<AvatarColor>({
      id: 33,
      nameKey: "Brown 2",
      hexColor: "#4b2a1a",
    });
    avatarEditorMock.pantsColor = new Observable<AvatarColor>({
      id: 33,
      nameKey: "Brown 2",
      hexColor: "#4b2a1a",
    });
    avatarEditorMock.shoesColor = new Observable<AvatarColor>({
      id: 33,
      nameKey: "Brown 2",
      hexColor: "#4b2a1a",
    });

    const { container } = render(
      <AvatarEditorClothingCategory
        viewModel={avatarEditorMock}
        controller={avatarEditorControllerMock}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  // ANF-ID: [EZZ0042, EZZ0043, EZZ0044]
  test.skip.each([
    ["shirt", true, false, false],
    ["pants", false, true, false],
    ["shoes", false, false, true],
  ])(
    "click on tile in category %s calls controller",
    (category, shirtUI, pantsUI, shoesUi) => {
      avatarEditorMock.uiVisiblity = {
        clothingMenu: {
          shirts: new Observable<boolean>(shirtUI),
          pants: new Observable<boolean>(pantsUI),
          shoes: new Observable<boolean>(shoesUi),
        },
      } as AvatarEditorUI;

      avatarEditorMock.shirt = new Observable<AvatarShirtModels>("none" as any);
      avatarEditorMock.pants = new Observable<AvatarPantsModels>("none" as any);
      avatarEditorMock.shoes = new Observable<AvatarShoesModels>("none" as any);
      avatarEditorMock.shirtColor = new Observable<AvatarColor>({
        id: 33,
        nameKey: "Brown 2",
        hexColor: "#4b2a1a",
      });
      avatarEditorMock.pantsColor = new Observable<AvatarColor>({
        id: 33,
        nameKey: "Brown 2",
        hexColor: "#4b2a1a",
      });
      avatarEditorMock.shoesColor = new Observable<AvatarColor>({
        id: 33,
        nameKey: "Brown 2",
        hexColor: "#4b2a1a",
      });

      const container = render(
        <AvatarEditorClothingCategory
          viewModel={avatarEditorMock}
          controller={avatarEditorControllerMock}
        />,
      );

      const tile = container.getByAltText("noneThumbnail");
      fireEvent.click(tile);

      expect(
        avatarEditorControllerMock.onAvatarConfigChanged,
      ).toHaveBeenCalled();
    },
  );
});
