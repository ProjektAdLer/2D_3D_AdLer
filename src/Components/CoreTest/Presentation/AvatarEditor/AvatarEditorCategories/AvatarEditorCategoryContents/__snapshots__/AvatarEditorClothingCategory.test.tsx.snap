// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarEditorClothingCategory should render 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      class="border-b border-gray-500 pb-2 mobile-landscape:ml-6"
    >
      <button
        class="w-full content-start text-left"
        title="accordionCategoryToolTip"
      >
        <h1
          class="text-lg font-bold xl:text-2xl"
        >
          ▼ shirtsTitle
        </h1>
      </button>
    </div>
    <div>
      <div
        class="accordionOpening"
      >
        <div
          class="grid gap-5 pt-4 rounded-lg p-4 portrait:grid-cols-3 grid-cols-5"
        >
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-topcloud"
              title="shirts-topcloud"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-topcloudThumbnail"
                  class="w-full h-full"
                  src="aa-none_Thumbnail.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-dress"
              title="shirts-dress"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-dressThumbnail"
                  class="w-full h-full"
                  src="aa-shirts-dress.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-hoodie"
              title="shirts-hoodie"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-hoodieThumbnail"
                  class="w-full h-full"
                  src="aa-shirts-hoodie.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-longdress"
              title="shirts-longdress"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-longdressThumbnail"
                  class="w-full h-full"
                  src="aa-shirts-longdress.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-sweatshirt"
              title="shirts-sweatshirt"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-sweatshirtThumbnail"
                  class="w-full h-full"
                  src="aa-shirts-sweatshirt.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-topcoatsanta"
              title="shirts-topcoatsanta"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-topcoatsantaThumbnail"
                  class="w-full h-full"
                  src="aa-shirts-topcoatsanta.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-topdresssanta"
              title="shirts-topdresssanta"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-topdresssantaThumbnail"
                  class="w-full h-full"
                  src="aa-shirts-topdresssanta.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shirts-vest"
              title="shirts-vest"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shirts-vestThumbnail"
                  class="w-full h-full"
                  src="aa-shirts-vest.png"
                />
              </div>
            </button>
          </div>
        </div>
        <div
          class="m-2 w-full p-2"
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <button
                class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
                title="ColorButtonToolTip"
              >
                <div
                  class="border-2 border-black rounded m-2 h-5 w-5"
                  style="background-color: rgb(75, 42, 26);"
                />
                Brown 2
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="border-b border-gray-500 pb-2 mobile-landscape:ml-6"
    >
      <button
        class="w-full content-start text-left"
        title="accordionCategoryToolTip"
      >
        <h1
          class="text-lg font-bold xl:text-2xl"
        >
          ▼ pantsTitle
        </h1>
      </button>
    </div>
    <div>
      <div
        class="accordionOpening"
      >
        <div
          class="grid gap-5 pt-4 rounded-lg p-4 portrait:grid-cols-3 grid-cols-5"
        >
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="pants-pantscloud"
              title="pants-pantscloud"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="pants-pantscloudThumbnail"
                  class="w-full h-full"
                  src="aa-none_Thumbnail.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="pants-cargo"
              title="pants-cargo"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="pants-cargoThumbnail"
                  class="w-full h-full"
                  src="aa-pants-cargo.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="pants-flared"
              title="pants-flared"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="pants-flaredThumbnail"
                  class="w-full h-full"
                  src="aa-pants-flared.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="pants-jeans"
              title="pants-jeans"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="pants-jeansThumbnail"
                  class="w-full h-full"
                  src="aa-pants-jeans.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="pants-santa"
              title="pants-santa"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="pants-santaThumbnail"
                  class="w-full h-full"
                  src="aa-pants-santa.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="pants-stockings"
              title="pants-stockings"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="pants-stockingsThumbnail"
                  class="w-full h-full"
                  src="aa-pants-stockings.png"
                />
              </div>
            </button>
          </div>
        </div>
        <div
          class="m-2 w-full p-2"
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <button
                class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
                title="ColorButtonToolTip"
              >
                <div
                  class="border-2 border-black rounded m-2 h-5 w-5"
                  style="background-color: rgb(75, 42, 26);"
                />
                Brown 2
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="border-b border-gray-500 pb-2 mobile-landscape:ml-6"
    >
      <button
        class="w-full content-start text-left"
        title="accordionCategoryToolTip"
      >
        <h1
          class="text-lg font-bold xl:text-2xl"
        >
          ▼ shoesTitle
        </h1>
      </button>
    </div>
    <div>
      <div
        class="accordionOpening"
      >
        <div
          class="grid gap-5 pt-4 rounded-lg p-4 portrait:grid-cols-3 grid-cols-5"
        >
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shoes-shoescloud"
              title="shoes-shoescloud"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shoes-shoescloudThumbnail"
                  class="w-full h-full"
                  src="aa-none_Thumbnail.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shoes-boots"
              title="shoes-boots"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shoes-bootsThumbnail"
                  class="w-full h-full"
                  src="aa-shoes-boots.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shoes-santaboots"
              title="shoes-santaboots"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shoes-santabootsThumbnail"
                  class="w-full h-full"
                  src="aa-shoes-santaboots.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shoes-santaheels"
              title="shoes-santaheels"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shoes-santaheelsThumbnail"
                  class="w-full h-full"
                  src="aa-shoes-santaheels.png"
                />
              </div>
            </button>
          </div>
          <div
            class="w-full h-full"
          >
            <button
              class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              data-testid="shoes-trainers"
              title="shoes-trainers"
            >
              <div
                class="flex flex-col"
              >
                <img
                  alt="shoes-trainersThumbnail"
                  class="w-full h-full"
                  src="aa-shoes-trainers.png"
                />
              </div>
            </button>
          </div>
        </div>
        <div
          class="m-2 w-full p-2"
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <button
                class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
                title="ColorButtonToolTip"
              >
                <div
                  class="border-2 border-black rounded m-2 h-5 w-5"
                  style="background-color: rgb(75, 42, 26);"
                />
                Brown 2
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
